src/A2A/Server/InMemoryTaskStore.cs

```diff
@@ -78,6 +80,16 @@ public Task SetTaskAsync(AgentTask task, CancellationToken cancellationToken = d
             return Task.FromCanceled(cancellationToken);
         }
 
+        if (task is null)
+        {
+            return Task.FromException(new ArgumentNullException(nameof(task)));
```

Comment:
Should we throw an A2AException with TaskNotFound, perhaps?
```suggestion
            return Task.FromException(new A2AException("Task not found", A2AErrorCode.TaskNotFound));
```

Reply:
I believe that not for `null`s, as when using `A2A.AspNetCore` we are never going to get a `null` for `taskId`. So the only scenario is somebody incorrectly using the library.

@brandonh-msft please mark as resolved if you agree